Source/Scene/PointCloud3DTileContent.js

```diff
@@ -823,58 +713,310 @@ define([
+            // Styling is handled in the batch table
+            hasStyle = false;
+        }
+
+        if (hasStyle) {
+            var shaderState = {
+                translucent : false
+            };
+            colorStyleFunction = style.getColorShaderFunction('getColorFromStyle', 'czm_tiles3d_style_', shaderState);
+            showStyleFunction = style.getShowShaderFunction('getShowFromStyle', 'czm_tiles3d_style_', shaderState);
+            styleTranslucent = shaderState.translucent;
+        }
+
+        content._styleTranslucent = styleTranslucent;
+
+        var hasColorStyle = defined(colorStyleFunction);
+        var hasShowStyle = defined(showStyleFunction);
+
+        // Get the properties in use by the style
+        var styleableProperties = [];
+        if (hasColorStyle) {
+            getStyleableProperties(colorStyleFunction, styleableProperties);
+        }
+        if (hasShowStyle) {
+            getStyleableProperties(showStyleFunction, styleableProperties);
+        }
+
+        // Disable vertex attributes that aren't used in the style, enable attributes that are
+        var styleableShaderAttributes = content._styleableShaderAttributes;
+        if (defined(styleableShaderAttributes)) {
+            for (name in styleableShaderAttributes) {
+                if (styleableShaderAttributes.hasOwnProperty(name)) {
+                    attribute = styleableShaderAttributes[name];
+                    var enabled = (styleableProperties.indexOf(name) >= 0);
+                    var vertexAttribute = getVertexAttribute(vertexArray, attribute.location);
+                    vertexAttribute.enabled = enabled;
+                }
+            }
+        }
+
+        if (hasColorStyle) {
+            // Disable the color vertex attribute when a color style is used
+            var colorVertexAttribute = getVertexAttribute(vertexArray, colorLocation);
+            colorVertexAttribute.enabled = false;
+        }
+
+        var attributeLocations = {
+            a_position : positionLocation
+        };
+        if (hasColors && !hasColorStyle) {
+            attributeLocations.a_color = colorLocation;
+        }
+        if (hasNormals) {
+            attributeLocations.a_normal = normalLocation;
+        }
+        if (hasBatchIds) {
+            attributeLocations.a_batchId = batchIdLocation;
+        }
+
+        var vs = 'attribute vec3 a_position; \n' +
+                 'varying vec4 v_color; \n' +
+                 'uniform float u_pointSize; \n' +
+                 'uniform vec4 u_highlightColor; \n';
+
+        var length = styleableProperties.length;
+        for (i = 0; i < length; ++i) {
+            name = styleableProperties[i];
+            attribute = styleableShaderAttributes[name];
+
+            //>>includeStart('debug', pragmas.debug);
+            if (!defined(attribute)) {
+                throw new DeveloperError('Style reference a property "' + name + '" that does not exist or is not styleable.');
```

Comment:
The spec and JS implementation return undefined. In GLSL we considered setting to a default value like 0 or keeping this `DeveloperError` in and instead checking during validation time.